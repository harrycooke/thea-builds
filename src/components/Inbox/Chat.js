import React from 'react';
import { connect } from 'react-redux';
import PropTypes from "prop-types";
import clsx from 'clsx';
import { readMessages, sendMessages } from "../../actions/chatActions";
import { modifyCase } from "../../actions/caseAction";
import { isEmpty, encodeString, decodeString } from "../../utils/functions";
import { withStyles } from '@material-ui/styles';
import _ from "../../@lodash";
import {
  Grid,
  Button,
  Typography,
  Paper,
  TextField,
  IconButton,
  Hidden,
  Divider,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  Slide,
  Avatar
} from "@material-ui/core";
import SendIcon from "@material-ui/icons/Send";
import PersonIcon from "@material-ui/icons/Person";

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});


const styles = theme => ({
  root: {
    width: "100%",
    height: "100%",
    padding: 10,
    marginBottom: 0
  },
  chattingBox: {
    width: "100%",
    height: "37vh",
    boxShadow: "none",
    overflow: 'auto',
  },
  chatSpecialist: {
    [theme.breakpoints.down('sm')]: {
      fontSize: 17,
      // fontWeight: 500,
    },
    [theme.breakpoints.up('sm')]: {
      fontSize: 25,
    },
     paddingTop: 0, 
     paddingLeft: 10
  },
  chatPatient: {
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
      // fontWeight: 900,
    },
    [theme.breakpoints.up('sm')]: {
      fontSize: 22,
    },
  },
  inputWrapper: {
    borderRadius: 24,
    height: "100%",
    backgroundColor: "#F1F1F1",
    boxShadow: "none",
    padding: theme.spacing(1, 2, 1, 2)
  },
  bottom: {
    width: "100%",
    padding: theme.spacing(1),
    marginTop: theme.spacing(2),
    backgroundColor: "#fff"
  },
  sendMessageItemWrapper: {
    display: "flex",
    float: "right",
    backgroundColor: "#EBF2FB",
    padding: 15,
    borderRadius: 10,
    margin: 10,
    maxWidth: 400,
    [theme.breakpoints.down('sm')]: {
      fontSize: 12,
    },
    [theme.breakpoints.up('sm')]: {
      fontSize: 14,
    },
  },
  receiveMessageItemWrapper: {
    display: "flex",
    float: "left",
    backgroundColor: "#F4F4F4",
    padding: 15,
    borderRadius: 10,
    margin: 10,
    maxWidth: 400,
    [theme.breakpoints.down('sm')]: {
      fontSize: 12,
    },
    [theme.breakpoints.up('sm')]: {
      fontSize: 14,
    },
  },
  buttonCenter:{
    margin: 'auto'
  }
})

export class Chat extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      messageText: '',
      chatList: [],
      channel_id: '',
      currentCase: '',
      clientdata: '',
      openAlert: false,
    };
    this.onMessageReceived = this.onMessageReceived.bind(this);
    this.scrollToBottom = this.scrollToBottom.bind(this);
  }

  UNSAFE_componentWillMount() {
    const { channel_id, currentCase } = this.props;
    this.setState({ channel_id, currentCase });
  }

  componentDidMount() {
    const { channel_id } = this.state;
    this.props.joinToRoom(channel_id);
    this.props.registerHandler(this.onMessageReceived);
    this.readMessage();
    this.scrollToBottom();
  }

  componentDidUpdate() {
    this.scrollToBottom();
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.currentCase) {
      const currentCase = nextProps.currentCase;
      this.setState({ currentCase });
    }
  }

  scrollToBottom = () => {
    this.panel.scrollTo(0, this.panel.scrollHeight);
  }

  readMessage = () => {
    const { channel_id } = this.state;
    const channelData = {
      channel_id: channel_id,
      type: "message"
    };
    this.props.readMessages(channelData);
  }

  onInputChange = (ev) => {
    this.setState({
      messageText: ev.target.value
    });
  }

  onMessageSubmit = (ev) => {
    ev.preventDefault();
    const { messageText, channel_id, currentCase } = this.state;
    if (messageText === '') {
      return;
    }
    
    //When there is no Specialty assigned to a case then display message to PCP
    /*if(!currentCase.senderData && (currentCase.caseData.specialist_id === "0" || !currentCase.caseData.specialist_id)){
      this.setState({
        messageText: '',
        openAlert: true
      });
      return;
    }*/

    
    let newMessageData = {
      channel_id: channel_id,
      sender_id: this.props.userdata.id,
      receiver_id: currentCase.senderData ? currentCase.senderData.id : 0,
      type: "message",
      content: encodeString(messageText),
      user_type: this.props.userdata.user_type,
      specialist_id: currentCase.caseData.specialist_id
    };
    this.setState((prevState) => ({
      chatList: [...prevState.chatList, newMessageData]
    }));
    this.props.sendMessage(channel_id, newMessageData);
    this.props.sendMessages(newMessageData);
    const current_time = new Date();
    const current_timestamp = current_time.getTime();
    let newCaseData = {
      caseId: channel_id,
      last_updated_by_id: this.props.userdata.id,
      is_read: true,
      updated_at: current_timestamp,
      questions: encodeString(messageText),
      is_deleted: false
    };
    
    if(this.props.userdata.user_type === 'PCP'){
      newCaseData['is_read'] = true;
      let caseData = currentCase.caseData, 
        readByOthers = caseData.is_read_by_other;
      if(readByOthers && (_.indexOf(readByOthers, caseData.specialist_id) >= 0)){
        readByOthers = _.pull(readByOthers, caseData.specialist_id);
        newCaseData['is_read_by_other'] = readByOthers;
      }
    }else if(this.props.userdata.user_type === 'Specialty'){
      newCaseData['is_read'] = false;
    }

    this.props.modifyCase(newCaseData);
    this.setState({
      messageText: ''
    });
  }
  
  handleClose = () => {
    this.setState({ openAlert: false });
  }

  onMessageReceived(entry) {
    this.updateChatHistory(entry);
  }

  updateChatHistory(entry) {
    this.setState((prevState) => ({
      chatList: [...prevState.chatList, entry]
    }));
  }

  render() {

    const { classes, readedMessages } = this.props;
    const { messageText, currentCase, openAlert } = this.state;
    let chatList;
    if (this.state.chatList === '') {
      chatList = readedMessages
    } else {
      chatList = readedMessages.concat(this.state.chatList)
    }
    
    const userAvatar = currentCase.senderData && currentCase.senderData.avatar ? currentCase.senderData.avatar.name : null ;
    
    return (
      <div className={classes.root}>
        {/* backgroundColor: "#aeaeae" */}
        <Grid container justify="flex-start" style={{ padding: 10, display: "flex", justifyContent: "flex-start", alignItems: "center" }}>
           {userAvatar ? <Avatar src={currentCase.senderData.avatar.name} /> : <PersonIcon style={{ marginTop: 0 }} />}
          {/* <PersonIcon style={{ marginTop: 8 }} /> */}
          {/* <Hidden xsDown> */}
            {!isEmpty(currentCase) && (
              <Typography className={classes.chatSpecialist}>
                {currentCase.caseData.specialty} Specialist
              </Typography>
            )}
          {/* </Hidden> */}
        </Grid>
        <Divider />
        <Grid container justify="center" style={{ height: "89%"}}>
          {!isEmpty(currentCase) && (
            <Typography className={classes.chatPatient}
              style={{ margin: 10, textAlign: "center" }}>
              {currentCase.caseData.patient_name} {currentCase.caseData.specialty} Consultation
            </Typography>
          )}
          <Paper className={classes.chattingBox} ref={(panel) => { this.panel = panel }}>
            {chatList.length > 0
              ? (
                <div>
                  {chatList.map((item, i) => {
                    if (String(item.sender_id) === this.props.userdata.id) {
                      return (
                        <div key={i} style={{ display: "inline-block", marginTop: 10, width: "100%" }}>
                          {item.content?
                            <div className={classes.sendMessageItemWrapper}>
                              <label style={{ marginTop: -38, marginLeft: -10, height: 20 }}>
                                Me
                              </label>
                              <Typography variant="inherit">
                                {decodeString(item.content)}
                              </Typography> 
                            </div>: null 
                          }
                        </div>
                      )
                    } 
                    else {
                      return (
                        <div key={i} style={{ display: "inline-block", marginTop: 10, width: "100%" }}>
                        {item.content?
                          <div className={classes.receiveMessageItemWrapper}>
                            {!isEmpty(currentCase) && (
                              <label style={{ marginTop: -38, marginLeft: -10, height: 20 }}>{/*currentCase.senderData? currentCase.senderData.lastName : null*/}{item.lastName}</label>
                            )}
                            <Typography variant="inherit">{decodeString(item.content)}</Typography>
                          </div>: null
                            }
                        </div>
                      )
                    }
                  })}
                </div>
              ) : (
                <div></div>
              )}
          </Paper>
          <form onSubmit={this.onMessageSubmit} className={clsx(classes.bottom, "py-16 px-8")}>
            <Grid container justify="center" spacing={0}>
              <Grid item xs={10} sm={11} md={11}>
                <Paper className={clsx(classes.inputWrapper, "flex items-center relative")}>
                  <TextField
                    autoFocus={false}
                    autoComplete="off"
                    id="message-input"
                    className={clsx(classes.typeYourMessage, "flex-1")}
                    className="flex-1"
                    InputProps={{
                      disableUnderline: true,
                      classes: {
                        root: "flex flex-grow flex-shrink-0 ml-16 mr-48 my-8",
                        input: ""
                      },
                      placeholder: "Type your message"
                    }}
                    InputLabelProps={{
                      shrink: false,
                      className: classes.bootstrapFormLabel
                    }}
                    onChange={this.onInputChange}
                    style={{ width: "100%" }}
                    value={messageText}
                    multiline
                    rows={4}
                    rowsMax={10}
                  />
                </Paper>
              </Grid>
              <Grid item xs={2} sm={1} md={1} style={{position: "relative"}}>
                <IconButton className="absolute right-0 top-0" type="submit" style={{position: "absolute", bottom: 0}}>
                  <SendIcon className="text-24" color="action" />
                </IconButton>
              </Grid>
            </Grid>
          </form>
        </Grid>
        <Dialog
          open={openAlert}
          TransitionComponent={Transition}
          keepMounted
          onClose={this.handleClose }
          aria-labelledby="alert-dialog-slide-title"
          aria-describedby="alert-dialog-slide-description"
          >
          <DialogContent>
            <DialogContentText id="alert-dialog-slide-description">
              No specialist assigned to this case yet. Please try later!
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClose} variant="contained" color="primary" className={classes.buttonCenter}>
              OK
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

Chat.propTypes = {
  readMessages: PropTypes.func.isRequired,
  sendMessages: PropTypes.func.isRequired,
  modifyCase: PropTypes.func.isRequired,
  userdata: PropTypes.object.isRequired,
  readedMessages: PropTypes.array.isRequired,
  currentCase: PropTypes.any.isRequired,
  errors: PropTypes.any.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata,
  readedMessages: state.chat.readMessages,
  currentCase: state.case.currentCase,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { readMessages, sendMessages, modifyCase }
)(withStyles(styles)(Chat));