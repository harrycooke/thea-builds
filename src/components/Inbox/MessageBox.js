import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({

});

export class MessageBox extends Component {
  render() {
    return (
      <div></div>
    );
  }
}

export default withStyles(styles)(MessageBox);