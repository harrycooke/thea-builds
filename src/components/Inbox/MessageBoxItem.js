import React, { Component } from 'react';
import { withStyles } from '@material-ui/styles';
import {
  Grid,
  Card,
  Typography,
  Avatar,
  Checkbox
} from "@material-ui/core"

const styles = theme => ({
  root: {
    padding: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },
  avatarWrap: {
    padding: 8
  }
})

export class MessageBoxItem extends Component {
  render() {
    const { classes } = this.props
    return (
      <Card className={classes.root}>
        <Grid container justify="center" spacing={1}>
          <Grid item xs={2} style={{ padding: 8 }}>
            <Avatar src={require("../../assets/images/alice.png")} />
          </Grid>
          <Grid item xs={7}>
            <Typography variant="h6">Dr. Anita Ramos</Typography>
            <label variant="subtitle1">Re. John Smith</label>
          </Grid>
          <Grid item xs={2}>
            <label>11:30am</label>
          </Grid>
          <Grid item xs={1}>
            <Checkbox color="primary" style={{ padding: 0 }} />
          </Grid>
        </Grid>
        <Typography color="textSecondary" style={{ marginTop: 10 }}>
          Looks like the patient is suffering from macular edema caused by their diabetes. Order OCT to confirm.
        </Typography>
      </Card>
    )
  }
}

export default withStyles(styles)(MessageBoxItem)