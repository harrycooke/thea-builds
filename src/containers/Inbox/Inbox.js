import React, { Fragment } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import MySnackbarContentWrapper from '../../components/SharedComponents/SnackBar';
import {
  Card,
  Grid,
  Typography,
  Button,
  Hidden,
  MenuItem,
  TextField,
  Divider,
  InputBase,
  InputAdornment,
  Snackbar,
} from '@material-ui/core';
import Media from 'react-media';
import SearchIcon from '@material-ui/icons/Search';
import AddIcon from "@material-ui/icons/Add";
import CreateIcon from '@material-ui/icons/Create';

import _ from "../../@lodash";
import { fetchAllCases, setCurrentCase, modifyCase, setSnackbar } from "../../actions/caseAction";
import { fetchAllDrafts, setCurrentDraft, removeDraft } from "../../actions/draftActions";
import MessageList from "../../components/Inbox/MessageList";
import styles from "../../assets/styles/InboxStyles";
import { specialtyList } from "../../utils/constant";
import ConfirmationDialog from "../../components/SharedComponents/ConfirmationDialog"
import { removeFileFromS3 } from '../../utils/s3';
import CircularProgress from '../../components/SharedComponents/CircularProgress';

import { sendAmplitudeData } from '../../utils/amplitude';

const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

class Inbox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      redirectToChat: false,
      redirectToAssignConsult: false,
      cases_id: null,
      specialty: null,
      redirectToCreateCase: false,
      readAllCases: [],
      readAllDrafts: [],
      selectedSpecialty: '',
      searchResultsForCases: [],
      searchResultsForDrafts: [],
      confirmDraftDelete: false,
      files: []
    }
    this.removeDraft = this.removeDraft.bind(this)
    this.openDraftDialog = this.openDraftDialog.bind(this)
    this.handleClose = this.handleClose.bind(this)
  }

  handleTrClicked = (id, isDraft, caseData) => {
    if (isDraft) {
      this.setState({
        redirectToCreateCase: true,
        selectedClientID: id
      });
      this.props.setCurrentDraft({ case_id: id })
    } else {
      let newCaseData = {
        caseId: id,
        last_updated_by_id: this.props.userdata.id
      }
      if(this.props.userdata.id === caseData.created_by_id){
        newCaseData['is_read'] = true;
      }else{
        let readByOthers = caseData.is_read_by_other ? caseData.is_read_by_other : [];
        let ifIndex = _.indexOf(readByOthers, this.props.userdata.id);
        if(ifIndex < 0){
          readByOthers.push(this.props.userdata.id);
          newCaseData['is_read_by_other'] = readByOthers;
        }
      }
      this.props.modifyCase(newCaseData);
      this.setState({
        redirectToChat: true,
        selectedClientID: id
      });
    }
  }

  handleSearch = e => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    })
    let selectedSpecialty = e.target.value;
    const { readAllCases, readAllDrafts } = this.state;
    if (selectedSpecialty !== '') {
      let searchResultsForCases = readAllCases;
      let searchResultsForDrafts = readAllDrafts;
      // debugger;
      if(selectedSpecialty != "All Specialties"){                                         //Need to bug-proof here. If the text value to if condition is changed, the condition will break and no value returned.
        searchResultsForCases = _.filter(readAllCases, function (item) {
          return item.caseData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.caseData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });
        searchResultsForDrafts = _.filter(readAllDrafts, function (item) {
          return item.draftData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.draftData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });  
      }
      this.setState({ searchResultsForCases, searchResultsForDrafts });
    } else {
    // else if (selectedSpecialty == '') {
    //   this.setState({
    //     searchResultsForCases: readAllCases,
    //     searchResultsForDrafts: readAllDrafts
    //   });
    // }
    //   else {
      this.setState({   
        searchResultsForCases: readAllCases,
        searchResultsForDrafts: readAllDrafts
      });
    }
  }

  componentDidMount() {
    const currentUserData = {
      currentUserId: this.props.userdata.id,
      currentUserRole: this.props.userdata.user_type,
      isDeleted: false,
      isCompleted: false
    }
    sendAmplitudeData('Open Inbox page', currentUserData)
    setTimeout(() => {
      this.props.fetchAllCases(currentUserData);
      this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
    }, 500)
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { readAllCases, readAllDrafts, errors } = nextProps;
    if (errors) {
      this.setState({
        errors: errors
      });
    }
    this.setState({
      readAllCases,
      readAllDrafts,
      searchResultsForCases: readAllCases,
      searchResultsForDrafts: readAllDrafts
    })
    if ((nextProps.isDraftDeleted !== this.props.isDraftDeleted) && nextProps.isDraftDeleted) {
      const currentUserData = {
        currentUserId: this.props.userdata.id,
        currentUserRole: this.props.userdata.user_type,
        isDeleted: false,
        isCompleted: false
      }
      setTimeout(() => {
        this.props.fetchAllCases(currentUserData);
        this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
      }, 500)
    }
  }
  removeDraft() {
    this.setState({ confirmDraftDelete: false })
    const { filesDraft, draftId } = this.state;
    if (filesDraft && filesDraft.length) {
      filesDraft.map(async (fileObj, index) => {
        if (fileObj && fileObj.name) {
          return await removeFileFromS3(fileObj, index);
        }
      })
    }
    this.props.removeDraft({ case_id: draftId })

  }
  handleClose() { this.setState({ confirmDraftDelete: false }) }

  handleSnackbarClose = () => { 
    this.props.setSnackbar(false);
  }

  openDraftDialog(draftData) {
    this.setState({ confirmDraftDelete: true, draftId: draftData.case_id, filesDraft: draftData.files && draftData.files.length ? draftData.files : [] })
  }

  handleAssignConsult = (id, specialty) => {
    if(id){
      this.setState({redirectToAssignConsult: true, cases_id: id, specialty: specialty});
    }
  }

  render() {
    const { classes, userdata, readAllDrafts } = this.props;
    const {
      redirectToChat,
      redirectToCreateCase,
      selectedClientID,
      selectedSpecialty,
      searchResultsForCases,
      searchResultsForDrafts,
      redirectToAssignConsult,
      cases_id,
      specialty
    } = this.state;
    if (redirectToChat) {
      return (<Redirect to={{
        pathname: `/insidemessage/room_${selectedClientID}`,
        state: { from: 'cases' }
      }} />)
    }
    if (redirectToCreateCase) {
      return (<Redirect to={{
        pathname: `/edit-new-case-${selectedClientID}`,
      }} />)
    }
    if(redirectToAssignConsult){
      return (<Redirect to={{
        pathname: `/assign-consult/case_${cases_id}/specialty_${specialty}`,
      }} />)
    }

    let rows = [...searchResultsForDrafts, ...searchResultsForCases];

    return (
      <div className={classes.container}>
        <CircularProgress isLoading={this.props.draftLoader} />
        <CircularProgress isLoading={this.props.caseLoader} />
        {/* 
        <div>
          <Media queries={{
            small: "(max-width: 599px)",
            medium: "(min-width: 600px) and (max-width: 1199px)",
            large: "(min-width: 1200px)"
          }}>
            {matches => (
              <Fragment>
                {matches.small && <p>I am small!</p> && <style>  </style>}
                {matches.medium && <p>I am medium!</p>}
                {matches.large && <p>I am large!</p>}
              </Fragment>
            )}
          </Media>
        </div> */}

        {/* <Hidden lgUp>
          <Grid container spacing={4} style={{ paddingBottom: 20 }}>
            <Grid item xs={5}
              style={{
                flexDirection: "row",
                display: "flex",
                justifyContent: "flex-start"
              }}
            >
              <Button
                color="primary"
                component={AdapterLink}
                to="/inbox/cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  width: 130,
                  // margin: 5
                }}>
                Open Cases
              </Button>
              <Button
                color="default"
                component={AdapterLink}
                to="/inbox/closed_cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  backgroundColor: "#ffffff",
                  width: 130,
                  marginLeft: 20,
                }}>
                Closed Cases
              </Button>
              {readAllDrafts && readAllDrafts.length ?
                <Button
                  color="default"
                  component={AdapterLink}
                  to="/inbox/drafts"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    backgroundColor: "#ffffff",
                    width: 80,
                    marginLeft: 32,
                  }}>
                  Drafts
              </Button> : null}
            </Grid>
            <Grid item xs={7}
              style={{
                display: "flex",
                justifyContent: "flex-end"
              }}
            >
              {userdata.user_type === "PCP" && (
                <Button
                  color="primary"
                  component={AdapterLink}
                  to="/create-new-case"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    marginLeft: 32,
                    height: 45
                  }}>
                  Create a new Case
              </Button>)}
            </Grid>
          </Grid>
        </Hidden> */}
        <Grid container justify="flex-start" spacing={0} className={classes.cardContainer} >
          <Grid item xs={12}>
            <Card style={{
              backgroundColor: "#ffffff",
              padding: 20,
              overflow: "auto",
              height: "100%"
            }}>
              <Grid  container justify="space-between" >
                <Grid style={{ paddingTop: 6, paddingLeft: 15, paddingRight: 15, paddingBottom: 20,
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"}}>
                  <Typography
                    color="textSecondary"
                    variant="h5"
                    fontWeight="100"
                  >
                    Open Cases
                  </Typography>
                  <Divider />
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4} 
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}>
                  <TextField
                    fullWidth
                    select
                    id="outlined-adornment-password"
                    className={classes.searchBar}
                    variant="outlined"
                    type={'text'}
                    size="small"
                    label="Filter by Specialty"
                    //label="Specialty type"
                    name={"selectedSpecialty"}
                    value={selectedSpecialty}
                    onChange={this.handleSearch}
                    style={{
                      display: userdata.user_type === "Admin" || userdata.user_type === "PCP"
                        ? "flex"
                        : "none"
                    }}
                  >
                    {specialtyList.map(specialty => (
                      <MenuItem key={specialty.value} value={specialty.value}>{specialty.label}</MenuItem>
                    ))}
                  </TextField>
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4}   
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}> 
                    <TextField
                      fullWidth
                      variant="outlined"
                      placeholder="Search"
                      size="small"
                      className={classes.searchBar}
                      InputProps={{
                       startAdornment: (
                         <InputAdornment position="start">
                           <SearchIcon  color="action"/>
                         </InputAdornment>
                       ),
                      }}
                      onChange={this.handleSearch}
                    />
                </Grid>
              </Grid>
              <MessageList
                openDraftDialog={this.openDraftDialog}
                rows={rows}
                isTrClicked={this.handleTrClicked}
                readAllCases={searchResultsForCases}
                readAllDrafts={searchResultsForDrafts}
                assignConsult={this.handleAssignConsult}
              />
            </Card>
            
          </Grid>
        </Grid>
        <ConfirmationDialog
          message="Are you sure you want to delete this draft?"
          title="Delete Draft"
          okOperationDialog={this.removeDraft}
          handleClose={this.handleClose}
          dialogState={this.state.confirmDraftDelete} />

          <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.props.showSnackbar}
            autoHideDuration={3000}
            onClose={this.handleSnackbarClose}
          >
            <MySnackbarContentWrapper
              onClose={this.handleSnackbarClose}
              variant="success"
              message="Case sent successfully."
            />
          </Snackbar>
      </div>
    );
  }
}

Inbox.propTypes = {
  userdata: PropTypes.object.isRequired,
  fetchAllCases: PropTypes.func.isRequired,
  setCurrentCase: PropTypes.func.isRequired,
  modifyCase: PropTypes.func.isRequired,
  fetchAllDrafts: PropTypes.func.isRequired,
  setCurrentDraft: PropTypes.func.isRequired,
  readAllCases: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata,
  readAllCases: state.case.readAllCases,
  readAllDrafts: state.draft.readAllDrafts,
  isDraftDeleted: state.draft.isDraftDeleted,
  draftLoader: state.draft.draftLoader,
  caseLoader: state.case.caseLoader,
  showSnackbar: state.case.showSnackbar
})

export default connect(
  mapStateToProps,
  { fetchAllCases, setCurrentCase, modifyCase, fetchAllDrafts, setCurrentDraft, removeDraft, setSnackbar }
)(withStyles(styles)(Inbox))