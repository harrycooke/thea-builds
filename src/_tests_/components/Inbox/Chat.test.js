import React from 'react';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6

import sinon from 'sinon';
import { mount } from 'enzyme';

import {Chat} from '../../../components/Inbox/Chat';

describe('Chat', () => {
    beforeAll = () => {
        this.state = {
            messageText: '',
            chatList: [],
            channel_id: '',
            currentCase: '',
            clientdata: '',
            openAlert: false,
          };
          this.onMessageReceived = this.onMessageReceived.bind(this);
          this.scrollToBottom = this.scrollToBottom.bind(this);
    };
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Chat, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
      const renderer = TestRenderer.create(Chat);
      renderer.toJSON();
    });
    

    describe('Chat', () => {
      it('should render correctly in "debug" mode', () => {
        const component = (Chat);
        expect(component).toMatchSnapshot();
      });
    });


    // it('calls funcName', () => {
    //     sinon.spy(Chat.prototype, 'funcName');
    //     const wrapper = mount(<Chat />);
    //     expect(Chat.prototype.componentDidMount).to.have.property('callCount', 1);
    //   });
    
      // it('allows us to set props', () => {
      //   const wrapper = mount(<Chat bar="baz" />);
      //   expect(wrapper.props().bar).to.equal('baz');
      //   wrapper.setProps({ bar: 'foo' });
      //   expect(wrapper.props().bar).to.equal('foo');
      // });
    
      // it('simulates click events', () => {
      //   const onButtonClick = sinon.spy();
      //   const wrapper = mount((
      //     <Chat onButtonClick={onButtonClick} />
      //   ));
      //   wrapper.find('button').simulate('click');
      //   expect(onButtonClick).to.have.property('callCount', 1);
      // });
});