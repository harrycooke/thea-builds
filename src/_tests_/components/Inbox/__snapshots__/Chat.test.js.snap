// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Chat should render correctly in "debug" mode 1`] = `[Function]`;
